src/Compilers/CSharp/Portable/Compilation/MethodBodySemanticModel.cs

```diff
@@ -224,6 +249,14 @@ internal override bool TryGetSpeculativeSemanticModelCore(SyntaxTreeSemanticMode
 
             var methodSymbol = (MethodSymbol)this.MemberSymbol;
             binder = new WithNullableContextBinder(SyntaxTree, position, binder);
+
+            // We don't need to loop over containing symbols chain.
```

Comment:
Not sure I understand why?

Reply:
@333fred We cannot get here with lambda or local function. https://github.com/dotnet/roslyn/pull/59767#discussion_r821154102

Reply:
Consider clarifying exactly that. And please leave a prototype comment to add an assert for this.

Reply:
No, I didn't see that comment in the incremental diff, that's fine. I would still like this comment itself to explain why we don't need to loop though.